### PR TITLE
Scripts for build and push images

### PR DESCRIPTION
**Pull request by phracek**
This PR moved scripts from `rhscl-container-ci` repo.
The original scripts are mentioned here:
https://github.com/sclorg/rhscl-container-ci/blob/master/yaml/builders/image-build-push.yaml

Signed-off-by: Petr "Stone" Hracek <phracek@redhat.com>

### CODE REVIEW

File: jenkins_ci/push-into-generated-branch.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+# Script pushes changes into generated branch if exists
+set -ex
+
+if git branch | grep -q generated; then
+  GIT_URL=$(git config --get remote.origin.url)
+  GIT_URL=${GIT_URL##git://}
+  git remote add auth_origin https://$GITHUB_PUSH_TOKEN@\$GIT_URL
```
Comment (pkubatrh) [inline jenkins_ci/push-into-generated-branch.sh:9]: Should the `$` over here be escaped? afaict `GIT_URL` will not be expanded that way

File: jenkins_ci/tag-push-to-registry.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# quay.io/centos7/python-38-centos7:1 quay.io/centos7/python-38-centos7:20201125-f9402d2 quay.io/centos7/python-38-centos7:latest quay.io/centos7/python-38-centos7:raw
+for image in $image_tags; do
+  case $image in
+    *:raw|*:squashed) continue ;;
+  esac
+
+  # By default images are now tagged into localhost/ registry
+  # Image name for Docker Hub
+  # hub_namespace is docker.io/centos and we need to get from image only image name
+  image_name={hub_namespace}/${image##*/}
```
Comment (pkubatrh) [inline jenkins_ci/tag-push-to-registry.sh:27]: Should `hub_namespace` be a variable? Is it set somewhere else?

File: jenkins_ci/push-into-generated-branch.sh (hunk 1 above)
Comment (phracek) [reply jenkins_ci/push-into-generated-branch.sh:9]: I think not. I have tried to set GITHUB_PUSH_TOKEN to some VALUE add command `git remote add auth_origin https://$GITHUB_PUSH_TOKEN@$GIT_URL` works fine.

Command `git remote -v` shows and output like:
```
auth_origin https://<VALUE>@GIT_URL (fetch)
auth_origin https://<VALUE>@GIT_URL (push)
```

But I am not 100% sure. Not a BASH expert;)

File: jenkins_ci/tag-push-to-registry.sh (hunk 2 above)
Comment (phracek) [reply jenkins_ci/tag-push-to-registry.sh:27]: This is valid, I will fix it.

File: jenkins_ci/push-into-generated-branch.sh (hunk 1 above)
Comment (pkubatrh) [reply jenkins_ci/push-into-generated-branch.sh:9]: Trying this out, with the escaped `$` I see as expected:
```
auth_origin	https://whatever@$GIT_URL (fetch)
auth_origin	https://whatever@$GIT_URL (push)
```

Without the escaped `$` I see:
```
auth_origin2	https://whatever@github.com:sclorg/s2i-python-container.git (fetch)
auth_origin2	https://whatever@github.com:sclorg/s2i-python-container.git (push)
```

So the question is, which one is it supposed to be?

Comment (phracek) [reply jenkins_ci/push-into-generated-branch.sh:9]: I guess the second one is correct.
See the output from Jenkins CI directly https://gist.github.com/rhscl-automation/750a6803eb1569633425a020af28cbb5#file-s2i-python-container-15-sh-L10740

```
+ git branch
+ grep -q generated
++ git config --get remote.origin.url
+ GIT_URL=git://github.com/sclorg/s2i-python-container.git
+ GIT_URL=github.com/sclorg/s2i-python-container.git
+ git remote add auth_origin https://****@github.com/sclorg/s2i-python-container.git
+ git push auth_origin generated
```

**Comment by pkubatrh** [comment]
LGTM now, thanks!

**Pull request merged**